eox_core/api/support/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,294 @@
+"""Support API integration tests."""
+
+from __future__ import annotations
+
+import ddt
+import requests
+from django.conf import settings as ds
+from django.urls import reverse
+from rest_framework import status
+
+from eox_core.api.v1.tests.integration.test_views import UsersAPIRequestMixin
+from eox_core.tests.integration.data.fake_users import FAKE_USER_DATA
+from eox_core.tests.integration.utils import BaseIntegrationTest, make_request
+
+settings = ds.INTEGRATION_TEST_SETTINGS
+
+
+class SupportAPIRequestMixin:
+    """Mixin class for the Pre Enrollments API request methods."""
+
+    DELETE_USER_URL = f"{settings['EOX_CORE_API_BASE']}{reverse('eox-support-api:eox-support-api:edxapp-user')}"
+    UPDATE_USERNAME_URL = (
+        f"{settings['EOX_CORE_API_BASE']}{reverse('eox-support-api:eox-support-api:edxapp-replace-username')}"
+    )
+    OAUTH_APP_URL = (
+        f"{settings['EOX_CORE_API_BASE']}{reverse('eox-support-api:eox-support-api:edxapp-oauth-application')}"
+    )
+
+    def delete_user(self, tenant: dict, params: dict | None = None) -> requests.Response:
+        """
+        Delete an edxapp user in a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            params (dict, optional): The query parameters for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "DELETE", url=self.DELETE_USER_URL, params=params)
+
+    def update_username(self, tenant: dict, params: dict | None = None, data: dict | None = None) -> requests.Response:
+        """
+        Update an edxapp user's username in a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            params (dict, optional): The query parameters for the request.
+            data (dict, optional): The body data for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "PATCH", url=self.UPDATE_USERNAME_URL, params=params, data=data)
+
+    def create_oauth_application(self, tenant: dict, data: dict | None = None) -> requests.Response:
+        """
+        Create an oauth application in a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            data (dict, optional): The body data for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "POST", url=self.OAUTH_APP_URL, json=data)
+
+
+@ddt.ddt
+class TestEdxAppUserAPIIntegration(SupportAPIRequestMixin, BaseIntegrationTest, UsersAPIRequestMixin):
```

Comment:
I know this PR is still a draft, so I’m holding off on a full review for now, but I think the formatting would look cleaner with:

```suggestion
class TestEdxAppUserAPIIntegration(
    SupportAPIRequestMixin,
    BaseIntegrationTest,
    UsersAPIRequestMixin,
):
```

Reply:
Done: https://github.com/eduNEXT/eox-core/commit/3ff25e8699849a907f7c6e184d18b8d7ee45e032